tests/integration/test_data/gomod_packages.yaml

```diff
@@ -0,0 +1,402 @@
+          replaces: null
+          type: go-package
+          version: null
+        - name: reflect
+          replaces: null
+          type: go-package
+          version: null
+        - name: rsc.io/quote
+          replaces: null
+          type: go-package
+          version: v1.5.2
+        - name: rsc.io/sampler
+          replaces: null
+          type: go-package
+          version: v1.3.0
+        - name: runtime
+          replaces: null
+          type: go-package
+          version: null
+        - name: runtime/internal/atomic
+          replaces: null
+          type: go-package
+          version: null
+        - name: runtime/internal/math
+          replaces: null
+          type: go-package
+          version: null
+        - name: runtime/internal/sys
+          replaces: null
+          type: go-package
+          version: null
+        - name: runtime/internal/syscall
+          replaces: null
+          type: go-package
+          version: null
+        - name: sort
+          replaces: null
+          type: go-package
+          version: null
+        - name: strconv
+          replaces: null
+          type: go-package
+          version: null
+        - name: strings
+          replaces: null
+          type: go-package
+          version: null
+        - name: sync
+          replaces: null
+          type: go-package
+          version: null
+        - name: sync/atomic
+          replaces: null
+          type: go-package
+          version: null
+        - name: syscall
+          replaces: null
+          type: go-package
+          version: null
+        - name: time
+          replaces: null
+          type: go-package
+          version: null
+        - name: unicode
+          replaces: null
+          type: go-package
+          version: null
+        - name: unicode/utf8
+          replaces: null
+          type: go-package
+          version: null
+        - name: unsafe
+          replaces: null
+          type: go-package
+          version: null
+        name: github.com/cachito-testing/cachito-gomod-with-deps
+        type: go-package
+        version: v0.0.0-20201001160613-4c65d49cae6b
+      - dependencies:
+        - name: golang.org/x/text
+          replaces: nulltest
+          type: gomod
+          version: v0.0.0-20170915032832-14c0d48ead0c
+        - name: rsc.io/quote
+          replaces: null
+          type: gomod
+          version: v1.5.2
+        - name: rsc.io/sampler
+          replaces: null
+          type: gomod
+          version: v1.3.0
+        name: github.com/cachito-testing/cachito-gomod-with-deps
+        type: gomod
+        version: v0.0.0-20201001160613-4c65d49cae6b
+  expected_files:
+    # app: https://github.com/cachito-testing/cachito-gomod-with-deps/tarball/4c65d49cae6bfbada4d479b321d8c0109fa1aa97
+    deps/gomod/pkg/mod/cache/download/rsc.io/quote/@v/v1.5.2.zip: https://github.com/cachito-testing/test_files/raw/master/test_gomod_with_deps_quote.tar.gz
```

Comment:
content was previously tested within app  part which is now missing should I try to add missing content for smapler and @v 
```
lkolacek@lkolacek-ThinkPad-T590 ~ $ tree /tmp/pytest-of-lkolacek/pytest-40/test_packages_gomod_packages_w0/gomod_packages-with_deps/
/tmp/pytest-of-lkolacek/pytest-40/test_packages_gomod_packages_w0/gomod_packages-with_deps/
├── deps
│   └── gomod
│       └── pkg
│           └── mod
│               └── cache
│                   └── download
│                       ├── golang.org
│                       │   └── x
│                       │       └── text
│                       │           └── @v
│                       │               ├── list
│                       │               ├── v0.0.0-20170915032832-14c0d48ead0c.info
│                       │               ├── v0.0.0-20170915032832-14c0d48ead0c.lock
│                       │               ├── v0.0.0-20170915032832-14c0d48ead0c.mod
│                       │               ├── v0.0.0-20170915032832-14c0d48ead0c.zip
│                       │               └── v0.0.0-20170915032832-14c0d48ead0c.ziphash
│                       └── rsc.io
│                           ├── quote
│                           │   └── @v
│                           │       ├── list
│                           │       ├── v1.5.2.info
│                           │       ├── v1.5.2.lock
│                           │       ├── v1.5.2.mod
│                           │       ├── v1.5.2.zip
│                           │       └── v1.5.2.ziphash
│                           └── sampler
│                               └── @v
│                                   ├── list
│                                   ├── v1.3.0.info
│                                   ├── v1.3.0.lock
│                                   ├── v1.3.0.mod
│                                   ├── v1.3.0.zip
│                                   └── v1.3.0.ziphash
└── output.json

```

Reply:
The missing zips for `rsc.io/sampler` and `golang.org/x/text`? +1 for those.

Do we report them in output.json properly?

Reply:
> Do we report them in output.json properly?

nevermind, I see them in the expected data :+1:

Reply:
Maybe we could also add a check to if all dependencies within expected_files match the ones we really have.